spec/datadog/profiling/native_extension_spec.rb

```diff
@@ -12,26 +12,90 @@
+          expect(error.instance_variable_get(:@telemetry_message)).to match(/a{255}\z/)
+        end
     end
 
     context "when called without releasing the gvl" do
+      it "raises a RuntimeError with appropriate error handling when called without GVL" do
+        expect do
+          described_class::Testing._native_grab_gvl_and_raise(ZeroDivisionError, "message %s", 'oops', false)
+        end.to raise_error(::RuntimeError) do |error|
+          expect(error.message).to include('grab_gvl_and_raise called by thread holding the global VM lock: message oops')
+          expect(error.instance_variable_get(:@telemetry_message)).to include('grab_gvl_and_raise called by thread holding the global VM lock: message %s')
+        end
+      end
+    end
+
+    context "when raising RuntimeError" do
+      subject(:raise_native_runtime_error) do
+        described_class::Testing._native_grab_gvl_and_raise(::RuntimeError, "runtime error test", nil, true)
+      end
+
       it "raises a RuntimeError" do
-        expect { described_class::Testing._native_grab_gvl_and_raise(ZeroDivisionError, "this is a test", nil, false) }
-          .to raise_exception(RuntimeError, /called by thread holding the global VM lock/)
+        expect { raise_native_runtime_error }.to raise_error(::RuntimeError) do |error|
+          expect(error.message).to eq("runtime error test")
+          expect(error.instance_variable_get(:@telemetry_message)).to eq("runtime error test")
+        end
+      end
+
+      it "is an instance of RuntimeError" do
+        expect { raise_native_runtime_error }.to raise_exception(RuntimeError)
+      end
+    end
+
+    context "when raising ArgumentError" do
+      subject(:raise_native_argument_error) do
+        described_class::Testing._native_grab_gvl_and_raise(::ArgumentError, "argument error test", nil, true)
+      end
+
+      it "raises an ArgumentError" do
+        expect { raise_native_argument_error }.to raise_error(::ArgumentError) do |error|
+          expect(error.message).to eq("argument error test")
+          expect(error.instance_variable_get(:@telemetry_message)).to eq("argument error test")
+        end
+      end
+
+      it "is an instance of ArgumentError" do
+        expect { raise_native_argument_error }.to raise_exception(ArgumentError)
+      end
+    end
+
+    context "when raising TypeError" do
+      subject(:raise_native_type_error) do
+        described_class::Testing._native_grab_gvl_and_raise(::TypeError, "type error test", nil, true)
+      end
+
+      it "raises a TypeError" do
+        expect { raise_native_type_error }.to raise_error(::TypeError) do |error|
+          expect(error.message).to eq("type error test")
+          expect(error.instance_variable_get(:@telemetry_message)).to eq("type error test")
+        end
+      end
+
+      it "is an instance of TypeError" do
+        expect { raise_native_type_error }.to raise_exception(TypeError)
       end
```

Comment:
Minor: The "is an instance of" tests can be removed, since the `raise_error(...)` validations in the "it raises ..." testcases already include that.

Reply:
Thank you! They are totally redundant!